docs/spec.md

```diff
@@ -538,6 +561,20 @@ All integer types explicitly specified (uint32, uint64, etc.) in this section ar
 | _encoding length_ | char[]  | encoding, same encoding that was used for the request |
 | remaining bytes   | uint8[] | response payload                                      |
 
+### Asset
+
+- Response to a [fetch asset](#fetch-asset) request if the asset was found
+- Only supported if the server previously declared the `assets` [capability](#server-info).
+
+| Bytes               | Type    | Description                                                                                            |
+| ------------------- | ------- | ------------------------------------------------------------------------------------------------------ |
+| 1                   | opcode  | 0x04                                                                                                   |
+| 4                   | uint32  | request id, as given in the corresponding [request](#fetch-asset)                                      |
+| 8                   | uint64  | Last modified time of the asset (nanoseconds)                                                          |
+| 4                   | uint32  | media type length                                                                                      |
+| _media type length_ | char[]  | [media type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) of the asset |
+| remaining bytes     | uint8[] | asset data (file contents)                                                                             |
```

Comment:
> How do you think it will work if the asset is very large? Will the websocket connection be able to handle it?

If the asset response message would exceed the max. websocket frame size, the message would be split up into multiple frames and send like that to the client. So there is theoretically no limit in file size

> I wonder if it would make sense to have an option where the server could return a url that the client can fetch() to load the asset in a separate request?

We could maybe use the `status` field for that:

```ts
enum FetchAssetStatus {
  SUCCESS_ASSET_DATA = 1,  // Request contains binary asset data
  SUCCESS_ASSET_URL = 2,    // Request contains asset URL
  // Other values: Error
}
```